opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -298,8 +302,8 @@ def __init__(
         self,
         trace_id: int,
         span_id: int,
-        trace_options: "TraceOptions" = None,
-        trace_state: "TraceState" = None,
+        trace_options: "TraceOptions" = DEFAULT_TRACE_OPTIONS,
+        trace_state: "TraceState" = DEFAULT_TRACE_STATE,
     ) -> None:
         if trace_options is None:
```

Comment:
We can remove this check then (and probably similar ones too), which is nice.

Reply:
yeah I thought about that, though it's possible someone could pass in `None` as a trace_options, decided to keep the check as a mechanisms against shooting yourself in the foot. I'm happy to take it out if you feel strongly about it, it would make the code simpler.

Reply:
I don't feel strongly about it. 🤷‍♂